stdlib/iarray.ml

```diff
@@ -169,57 +392,137 @@ let fold_right f a x =
   done;
   !r
 
+(* CR aspectorzabusky: Why do I need this?  Shouldn't mode-crossing handle doing
+   this? *)
+let[@inline always] globalize_bool : local_ bool -> bool = fun b -> b
+
 let exists p a =
   let n = length a in
-  let rec loop i =
+  let rec loop i = local_
```

Comment:
I don't think you really need the `local_` here, or in the `local` version. Ditto similar functions below.

Reply:
I'm fairly sure I only added that as requested but I can double-check.